std/experimental/xml/appender.d

```diff
@@ -0,0 +1,198 @@
+/*
+*             Copyright Lodovico Giaretta 2016 - .
+*  Distributed under the Boost Software License, Version 1.0.
+*      (See accompanying file LICENSE_1_0.txt or copy at
+*            http://www.boost.org/LICENSE_1_0.txt)
+*/
+
+/*
+*   Appender implementation that uses a custom allocator. Meant for internal usage.
+*/
+
+module std.experimental.xml.appender;
+
+/*
+*   Appender implementation that uses a custom allocator. Meant for internal usage.
+*/
+package struct Appender(T, Alloc)
+{
+    import std.experimental.allocator;
+    import std.array;
+    import std.range.primitives;
+    import std.traits;
+
+    Alloc* allocator;
+    private Unqual!T[] arr;
+    private size_t used;
+
+    public this(ref Alloc alloc)
```

Comment:
This is trivial but you have an inconsistent usage of `public` here and in 3 next methods. They would only be justified after a global `private:` since public is the default protection attribute.